### ISSUE TITLE
Significance of line 174 in train_query.py code 

### ISSUE THREAD

**Issue by Nishant3815** [princeton-nlp/DensePhrases#24]
Hi, 

I was going through the code for query finetuning and I am not able to understand one condition in the code:

![image](https://user-images.githubusercontent.com/17989690/161414268-ecbf4e58-4728-4195-a0c8-53f60fe14a96.png)

Is the above highlighted line redundant and if not what is the significance (I feel we can directly update the encoder). Just wanted to make sure that I am not missing anything.

**Comment by jhyuklee** [comment]
Hi @Nishant3815,

this line (v1.0.0) was written to sync the pre-trained encoder (for the retrieval) with the target encoder that is used for the query-side fine-tuning. As you mentioned, we updated the code (v1.1.0) recently to directly update the target encoder and use it for the retrieval as well: https://github.com/princeton-nlp/DensePhrases/blob/v1.1.0/train_query.py

It gives a slightly better accuracy, overall.

**Comment by Nishant3815** [comment]
Thanks for the confirmation on updating the target encoder.  This makes sense, I had been following a previous version. 

For your comment, "this line (v1.0.0) was written to sync the pre-trained encoder (for the retrieval) with the target encoder that is used for the query-side fine-tuning", in a scenario where we would like to keep both pretrained_encoder and target_encoder version of the code, can we keep line 175 and 176 and remove the "if" statement of 174.

**Comment by Nishant3815** [comment]
Thanks, this helps.
